java/ext/javadoctest.py

```diff
@@ -0,0 +1,92 @@
+import os
+import pathlib
+import subprocess
+
+from sphinx.ext.doctest import (Any, Dict, DocTestBuilder, TestcodeDirective,
+                                TestoutputDirective, doctest, sphinx)
+from sphinx.locale import __
+
+
+class JavaDocTestBuilder(DocTestBuilder):
+    """
+    Runs java test snippets in the documentation.
+    """
+
+    name = "javadoctest"
+    epilog = __(
+        "Java testing of doctests in the sources finished, look at the "
+        "results in %(outdir)s/output.txt."
+    )
+
+    def compile(
+        self, code: str, name: str, type: str, flags: Any, dont_inherit: bool
+    ) -> Any:
+        # go to project that contains all your arrow maven dependencies
+        path_arrow_project = os.path.join(pathlib.Path.cwd(), "source", "demo")
+
+        # create list of all arrow jar dependencies
+        subprocess.check_call(
+            [
+                "mvn",
+                "-q",
+                "dependency:build-classpath",
+                "-DincludeTypes=jar",
+                "-Dmdep.outputFile=.cp.tmp",
+            ],
+            cwd=path_arrow_project,
+            text=True,
+        )
+        if not os.path.exists(path_arrow_project + "/.cp.tmp"):
+            raise RuntimeError(
+                __("invalid process to create jshell dependencies library")
+            )
+
+        # get list of all arrow jar dependencies
+        with open(os.path.join(path_arrow_project, ".cp.tmp")) as f:
+            stdout_dependency = f.read()
+        if not stdout_dependency:
+            raise RuntimeError(
+                __("invalid process to list jshell dependencies library")
+            )
+
+        # execute java testing code thru jshell and read output
+        proc_jshell_process = subprocess.Popen(
+            ["jshell", "--class-path", stdout_dependency, "-"],
+            stdin=subprocess.PIPE,
+            stdout=subprocess.PIPE,
+            text=True,
+        )
+        out_java_arrow, err_java_arrow = proc_jshell_process.communicate(code)
+        if err_java_arrow:
+            raise RuntimeError(__("invalid proc ess to run jshell"))
+
+        # continue with python logic code to do java output validation
+        output = "print('" + self.clean_output(out_java_arrow) + "')"
```

Comment:
I see, this is a smart trick to reuse the existing doctest logic.
My suggestion would be to turn this to a f-string and use `'''` to delimit the string so you don't have to deal with newlines.

Something like

```
output = f"print('''{out_java_arrow}'''")
```

(You would probably still have to apply `clean_output` for tabs btw.

Reply:
This PR is to handle java cookbook for create arrow object.

This recommendation was added to baseline PR: https://github.com/apache/arrow-cookbook/pull/125/commits/3d82096ade13f80fbc37be274efc7b4f1b0712c4

Please if we could merge that to continue with another cookbooks

Reply:
Please if we could merge that to continue with another cookbooks: https://github.com/apache/arrow-cookbook/pull/125